src/DCDataSet.cpp

```diff
@@ -36,13 +38,52 @@
 
 namespace splash
 {
+    herr_t H5Ewalk_strings(unsigned n, const H5E_error2_t *err_desc, void *client_data)
+    {
+        std::ostringstream *stream = (std::ostringstream *)client_data;
+        const char		*maj_str = NULL;
+        const char		*min_str = NULL;
+
+        /* Check arguments */
+        assert (err_desc);
+        if (!client_data) client_data = stderr;
+
+        /* Get descriptions for the major and minor error numbers */
+        maj_str = H5Eget_major (err_desc->maj_num);
+        min_str = H5Eget_minor (err_desc->min_num);
+
+        /* Print error message */
+        
+        *stream << "  #" << std::setfill('0') << std::setw(3) << n;
+        *stream << ": " <<  err_desc->file_name << " line " << err_desc->line;
+        *stream << " in " << err_desc->func_name << "(): " << err_desc->desc << std::endl;
+        
+        *stream << "    major(" << std::setfill('0') << std::setw(2) << err_desc->maj_num;
+        *stream << "): " << maj_str << std::endl;
+        
+        
+        *stream << "    minor(" << std::setfill('0') << std::setw(2) << err_desc->min_num;
+        *stream << "): " << min_str << std::endl;
+        
+        return 0;
+    }
+
+    std::string DCDataSet::getHDF5ExceptionString(std::string msg)
```

Comment:
We need to distinguish between exceptions that are triggered by errors in HDF5-functions and other exceptions that are for example triggered by a bool-variable which signals that a certain file is/is not open. In this way we avoid showing HDF5-errors when there are none. (Otherwise the `DEBUG_VERBOSE` approach would suffice.) Towards solving this issue for the entire `splash` codebase, this need for distinction between HDF5 and non-HDF5 related exceptions also means that we need to go through all exceptions in the code (which is some work).